### PR TITLE
Promote node to leader

### PR DESCRIPTION
**Pull request by andre8244**
Replaces https://github.com/NethServer/ns8-core/pull/378

**Nodes** page: promote node as leader role

Other changes:
- **App instances page**: show latest version available for upgradable apps
- Minor improvements

promote1

promote2

promote3

### CODE REVIEW

File: core/ui/src/views/Nodes.vue [hunk 1]
```diff
@@ -309,7 +334,7 @@ export default {
   },
   data() {
     return {
-      REFRESH_DATA_TIME_INTERVAL: 5000,
+      REFRESH_DATA_TIME_INTERVAL: 10000, //// 5000
```
Comment (gsanchietti) [inline core/ui/src/views/Nodes.vue:337]: Maybe this one should be reverted?

File: core/ui/public/i18n/en/translation.json [hunk 2]
```diff
@@ -1034,7 +1037,27 @@
...
+    "remove_node_apps_confirm": "The following app is installed on {node}: <strong>{nodeApps}</strong>.<br>If you remove the node, <strong>{nodeApps} will be removed as well</strong>. | The following apps are installed on {node}: <strong>{nodeApps}</strong>.<br>If you remove the node, <strong>these apps will be removed as well</strong>.",
+    "promote_to_leader": "Promote to leader",
+    "promote_node_to_leader": "Promote node to leader",
+    "promote_node_confirm": "<strong>{name}</strong> will be promoted to leader role.",
+    "type_to_confirm_promotion": "Type <strong>{name}</strong> to confirm node promotion",
+    "node_vpn_endpoint_address": "{node} VPN endpoint address",
+    "node_vpn_endpoint_address_tooltip": "Make sure this VPN endpoint address is reachable by all other cluster nodes",
+    "node_vpn_endpoint_port": "{node} VPN endpoint port",
+    "understood_promote_node": "I understand, promote node",
+    "reset_backup_encryption_password_info": "After node promotion, go to <strong>Backup</strong> section and reset backup encryption password",
```
Comment (gsanchietti) [inline core/ui/public/i18n/en/translation.json:1050]:
```suggestion
    "reset_backup_encryption_password_info": "After node promotion, go to <strong>Backup</strong> section and setup the backup encryption password",
```

File: core/ui/public/i18n/en/translation.json [hunk 3]
```diff
@@ -1034,7 +1037,27 @@
...
+    "new_cluster_admin_message": "From now on, cluster administration will be available at:<br><code>https://{endpointAddress}/cluster-admin/</code>",
+    "check_node_connectivity": "Check node connectivity",
+    "cannot_promote_node": "Cannot promote node",
+    "validation_node_is_already_leader": "Node already has leader role",
+    "validation_endpoint_address_not_reachable": "VPN endpoint address is not reachable, please check it again. If you know what you are doing and want to skip this check, deselect <strong>Check node connectivity</strong> and try again",
```
Comment (gsanchietti) [inline core/ui/public/i18n/en/translation.json:1055]: This is a check box, maybe `uncheck` is better?
```suggestion
    "validation_endpoint_address_not_reachable": "VPN endpoint address is not reachable, please check it again. If you know what you are doing and want to skip this check, uncheck <strong>Check node connectivity</strong> and try again",
```

File: core/ui/src/views/Nodes.vue (hunk 1 above)
Comment (andre8244) [reply core/ui/src/views/Nodes.vue:337]: Yes :cry:

**Review by gsanchietti — APPROVED** [review]

**Review by gsanchietti — APPROVED** [review]

**Review by edospadoni — APPROVED** [review]

**Pull request merged**